src/sage/crypto/public_key/key_exchange/commutative_key_exchange.py

```diff
@@ -0,0 +1,35 @@
+from sage.crypto.public_key.key_exchange import KeyExchange
+from abc import ABC, abstractmethod
```

Comment:
As far as I can tell this would be the first use of `ABC` or `abstractmethod` in Sage. Sage has its own `abstract_method`. I see no reason to use that here, and am more than happy to use the standard Python `abc` library. My best guess is the custom one is to better support the category stuff, but we don't need that here.